src/main/java/frc/robot/Robot.java

```diff
@@ -6,139 +6,137 @@
-   * This function is called periodically during autonomous.
-   */
+/**
+* This function is called periodically during autonomous.
+*/
   @Override
   public void autonomousPeriodic() {
 
   }
 
-  /**
-   * This function is called once each time the robot enters teleoperated mode.
-   */
+/**
+* This function is called once each time the robot enters teleoperated mode.
+*/
   @Override
   public void teleopInit() {
   }
-
-  /**
-   * This function is called periodically during teleoperated mode.
-   */
+  
+/**
+* This function is called periodically during teleoperated mode.
+*/
   @Override
   public void teleopPeriodic() {
 
-   //PIDController pid = new PIDController(kP, kI, kD); // This was for a pid that is currently not finished...
-   //pid.setTolerance(5, 10);
-   //pid.atSetpoint();
-   //pid.setIntegratorRange(-0.5, 0.5);
-   //pid.enableContinuousInput(-180, 180);
-   //SmartDashboard.putNumber("PID Graph Test", (pid.calculate(encoder.getDistance(), setpoint)));
-
-   boolean buttonValueA; // A is 1
-   boolean buttonValueB; // B is 2
-   boolean buttonValueX; // X is 3
-   boolean buttonValueY; // Y is 4
-   
-   SmartDashboard.putNumber("Joystick X Value", joystick.getX());
-   SmartDashboard.putNumber("Joystick Y Value", joystick.getY()* -1);
-   SmartDashboard.putBoolean("X Too Slow?", joystick.getX() < -0.9);
-   SmartDashboard.putBoolean("X Too Fast?", joystick.getX() > 0.9);
-   SmartDashboard.putBoolean("Y Too Fast?", joystick.getY() < -0.9);
-   SmartDashboard.putBoolean("Y Too Slow?", joystick.getY() > 0.9);
-
-  if (joystick.getX() < -0.9) {
-    joystick.setRumble(RumbleType.kLeftRumble, 1);
-    joystick.setRumble(RumbleType.kRightRumble, 1);
-    }
-    else {
-    joystick.setRumble(RumbleType.kLeftRumble, 0);
-    joystick.setRumble(RumbleType.kRightRumble, 0);
-  }
-
-  if (joystick.getX() > 0.9) {
-    joystick.setRumble(RumbleType.kLeftRumble, 1);
-    joystick.setRumble(RumbleType.kRightRumble, 1);
+    // shift alt F
+
+    if (SecondsLeft > 0) {
+      SmartDashboard.putNumber("Timer", m_timer.get());
+      if (m_timer.get() > 1) {
+        m_timer.stop();
+        m_timer.reset();
+        SmartDashboard.putNumber("Time Left", SecondsLeft);
+        SecondsLeft = SecondsLeft - 1;
+        m_timer.start();
+      }
+      SmartDashboard.putBoolean("ENDGAME", SecondsLeft < 30);
+      SmartDashboard.putBoolean("GAME OVER", SecondsLeft < 1);
+
+      boolean buttonValueA; // A is 1
+      boolean buttonValueB; // B is 2
+      boolean buttonValueX; // X is 3
+      boolean buttonValueY; // Y is 4
+
+      double XvalueL = joystick.getRawAxis(0);
+      double YvalueL = joystick.getRawAxis(1);
+      double XvalueR = joystick.getRawAxis(4);
+      double YvalueR = joystick.getRawAxis(5);
+
+      // Left Joystick
+      SmartDashboard.putNumber("Joystick XL Value", XvalueL);
+      SmartDashboard.putNumber("Joystick YL Value", YvalueL * -1);
+      SmartDashboard.putBoolean("XL Too Slow?", XvalueL < -0.9);
```

Comment:
What does XL and YL stand for? 
When numbers appear in code that look measured or arbitrary we call them "magic numbers" since their meaning and how they were derived are a mystery to anyone outside of the author. One way to make them more understandable is to assign the number to a descriptive variable name. e.g. public static final double <MEANINGFUL_NAME_OF_VARIABLE> = 0.9

Reply:
Oh I understand. XL meant the x value of the Left joystick, YR is the y value of the Right, and so on... I will make sure to put a better name for the variables.